aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1434,23 +1443,16 @@ Tensor amin(const Tensor& self, IntArrayRef dim, bool keepdim) {
   return at::amin_out(result, self, dim, keepdim);
 }
 
-Tensor &amax_out(const Tensor& self, IntArrayRef dim, bool keepdim, Tensor& result) {
+TORCH_IMPL_FUNC(amax_out) (const Tensor& self, IntArrayRef dim, bool keepdim, const Tensor& result) {
   TORCH_CHECK(self.scalar_type() == result.scalar_type(), "Expected the dtype for input and out to match, but got ",
-              self.scalar_type(), " for input's dtype and ",  result.scalar_type(), " for out's dtype.");
-  if (self.numel() == 0) {
-    zero_numel_check_dims(self, dim, "amax()");
-  }
+            self.scalar_type(), " for input's dtype and ",  result.scalar_type(), " for out's dtype.");
+  c10::MaybeOwned<Tensor> in = c10::MaybeOwned<Tensor>::borrowed(self);
```

Comment:
Just checking since I'm not sure if this is necessary - is there a reason you added the `MaybeOwned` call here, instead of passing `self` directly into `meta::make_reduction`?

Reply:
bump on this - there are actually some docs on `MaybeOwned` [here](https://pytorch.org/cppdocs/notes/maybe_owned.html). I don't think you need it in this case (you should be able to get away with passing in `self` directly to `make_reduction()`).

It's mostly an optimization that you can use for performance, when you know that in some cases you have the only reference to the tensor. In this particular case you can't ever guarantee that, since `self` just came directly from code outside of the kernel.

There's an example of its usage in the same file [here](https://github.com/pytorch/pytorch/blob/5e6f29661254e5ebc97fac3d829e5d455cda5864/aten/src/ATen/native/ReduceOps.cpp#L1479) if you're curious - in that line, by doing the reshape() we construct a new tensor, which we *know* has no outstanding references since we created it on the spot. We can use that to squeeze out a bit of perf by using `MaybeOwned::owned`.